### ISSUE TITLE
NVIDIA - N64 Depth Compare Problems

### ISSUE THREAD

**Issue by Predator82Germany** [gonetz/GLideN64#332]
N64 depth compare didn´t work (All Regions)

Without: https://www.youtube.com/watch?v=8hrpB9s-UtU
With https://www.youtube.com/watch?v=RZoE8Jd2Jsc

We discussed a couple of weeks ago on the board but not if it´s fixable

**Comment by olivieryuyu** [comment]
i don't have this issue. It works for me.

Do you use the ini settings? If so you need to deactivate it and then set the N64 depth compare option

**Comment by Predator82Germany** [comment]
Is without *.ini

**Comment by olivieryuyu** [comment]
nvidia or amd?

**Comment by Predator82Germany** [comment]
NVIDIA

**Comment by olivieryuyu** [comment]
AMD on my side and Gonetz too. Possibly the root of the issue.

**Comment by gonetz** [comment]
Yes, current implementation of that shader is incompatible with NVidia. I need NVidia card to fix that. I can try to fix it with "guess and try" method, but it will take time.

**Comment by Mushman** [comment]
N64 Depth Compare is causing problems in other games on NVidia hardware too.

Mario Golf lens flare effect should have disappeared at this angle
![Enabling N64 Depth Compare actually disables some depth effects](https://cloud.githubusercontent.com/assets/6550546/6891838/252ac5fa-d70a-11e4-9613-8ee5d4f4eef0.png)

Jet Force Gemini also has some bugs with it on
![The star's light clearly should hidden in this shot](https://cloud.githubusercontent.com/assets/6550546/6891861/4ab20996-d70a-11e4-9bde-1c60e4cb0273.png)

![Garbage from the waterfall on the other side of the outcrop](https://cloud.githubusercontent.com/assets/6550546/6891886/6c3b5a2c-d70a-11e4-9449-ce3d73b34a3b.png)

**Comment by gonetz** [comment]
Need to find, is it never working on NVidia, or I broke it at some moment.
Please test New Year Beta build, does it work with NVidia?

**Comment by Predator82Germany** [comment]
New Year Beta have this problems too nut tested only Extreme-G

**Comment by baptiste0602** [comment]
I confirm, i tested with JFG

**Comment by gonetz** [comment]
Hmm. may be it works with first monthly beta? Or with the first weekly beta?

**Comment by Predator82Germany** [comment]
Didn´t

**Comment by Mushman** [comment]
The issue is present in the first release of GLideN64. It is not a regression.

**Comment by gonetz** [comment]
So, I have no starting point :(

**Comment by Predator82Germany** [comment]
& the source plugin?

**Comment by gonetz** [comment]
What?

**Comment by Predator82Germany** [comment]
you started GLideN64 with a source?

**Comment by Mushman** [comment]
I think he's asking whether it occurs in glN64.

**Comment by Predator82Germany** [comment]
Yeah

**Comment by Predator82Germany** [comment]
Yeah

**Comment by gonetz** [comment]
glN64 does not use shaders at all

**Comment by gonetz** [comment]
I'll prepare test build, similar to cel-shading one. That is the problem shader will be loaded from a text file. May be somebody will find a bug in it.

**Comment by Predator82Germany** [comment]
See no difference

**Comment by gonetz** [comment]
Uploaded. Instruction: 
- unpack dll and shader files to Mupen64Plus folder 
- test rom is Extreme-G, extrem-g logo should look like on my screenshot.
- run rom. Check gliden64.log for "Image textures supported" string. If you see "Image textures NOT supported " don't even try to test.
- depthcompare.shader is my current version of that shader. It obviously will not work for you. In 'Shader' folder you will find several 'Hack' subfolders with simplified version of the shader. Please try them one by one instead of the default shader.
- If you know GLSL, you may try to find a bug in my code and fix it.

**Comment by gonetz** [comment]
![extremeg-000](https://cloud.githubusercontent.com/assets/8983092/7101682/8ea34ad6-e086-11e4-942e-348ee9e956f4.png)

**Comment by gonetz** [comment]
OpenGL version: 3.3.0 ?!
No more questions. Depth compare shader requires OpenGL 4.3

**Comment by Predator82Germany** [comment]
Then depth died for me til new gfx?

**Comment by gonetz** [comment]
You can't use N64 depth compare option. No need to set it off in the ini file, it will be disabled automatically.

**Comment by Mushman** [comment]
Until now, I have not tested with Extreme-G. I was testing with Mario Golf and Jet Force Gemini. When I commented saying the issue was present in the first release of GLideN64, I did not test with Extreme-G.

I tested the Depth Compare build and the hacked shaders with Mario Golf and none solved the issue. However, I then tested Extreme-G with the original shader and it worked fine. Testing the latest Mupen release also works fine.

While the error is still present, I am sorry for not testing Extreme-G as I was expected to. I will go back and regression test with Extreme-G and report back with the results.

For the record, I am testing with an NVidia Geforce GTX 970 with OpenGL 4.5.0 and gliden64.log reports that images textures are supported.

**Comment by gonetz** [comment]
Current version of N64 depth compare shader is imprecise, imperfect and incomplete. Complete N64 depth compare impossible with pixel coverage emulation. Thus, currently that option is useful only for a few games. Extreme-G and Harvest Moon among them.

**Comment by Predator82Germany** [comment]
Is a OpenGL 4.4 supported card enough for GLideN64?

**Comment by gonetz** [comment]
My graphics card is 4.4 compatible. Not sure that 4.5 will be available for it

**Comment by Mushman** [comment]
Extreme-G's depth effect works in the first weekly release of GLideN64 and all other versions I tested with NVidia hardware. The problems with Mario Golf and Jet Force Gemini remain with the option selected, but those games might not need it.

I'm sorry for commenting with the screenshots for Mario Golf and Jet Force Gemini. I presumed that the original case for the issue was valid and that they too might help solve it. It turns out that this issue was user error, and that my contribution only confused the situation.

EnableN64DepthCompare should be disabled for both Mario Golf and Jet Force Gemini in the game configuration. Unless the setting is required for a game to display effects but still causes errors on NVidia hardware, this issue is not valid.
